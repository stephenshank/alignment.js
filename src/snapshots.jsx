import React from "react";

import fastaParser from "./helpers/fasta";
import { BaseSVGAlignment } from "./components/BaseAlignment";

const fasta = `
>Seq1
ATCGTAATTGCA
>Seq2
CTCGTAATGGCC
>Seq3
GTCGTCAATGCT
`,
  parsed_fasta = fastaParser(fasta);

function VanillaBaseSVGAlignment() {
  return (
    <svg>
      <BaseSVGAlignment sequenceData={parsed_fasta} />
    </svg>
  );
}

function AllSnapshots() {
  return (
    <div>
      <h4>VanillaBaseSVGAlignment</h4>
      <VanillaBaseSVGAlignment />
    </div>
  );
}

export default AllSnapshots;
export { VanillaBaseSVGAlignment };
