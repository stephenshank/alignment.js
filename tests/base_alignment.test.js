const fs = require("fs");
import React from "react";
import renderer from "react-test-renderer";

import { VanillaBaseSVGAlignment } from "../lib/snapshots";

test("BaseSVGAlignment renders correctly with vanilla options", () => {
  const fasta = fs.readFileSync("./dist/data/Simple.fasta"),
    vanilla_base_svg_alignment = renderer
      .create(<VanillaBaseSVGAlignment />)
      .toJSON();
  expect(vanilla_base_svg_alignment).toMatchSnapshot();
});
