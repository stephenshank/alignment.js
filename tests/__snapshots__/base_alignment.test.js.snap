// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseSVGAlignment renders correctly with vanilla options 1`] = `
<svg>
  <g
    style={
      Object {
        "fontFamily": "Courier",
        "height": 60,
        "width": 240,
      }
    }
    transform="translate(0,0)"
  >
    <g
      transform="translate(0, 0)"
    >
      <rect
        fill="LightPink"
        height={21}
        width={21}
        x={0}
        y={0}
      />
      <text
        dy=".25em"
        fill="Red"
        textAnchor="middle"
        transform="translate(10, 10)"
      >
        A
      </text>
    </g>
    <g
      transform="translate(20, 0)"
    >
      <rect
        fill="LightBlue"
        height={21}
        width={21}
        x={0}
        y={0}
      />
      <text
        dy=".25em"
        fill="Blue"
        textAnchor="middle"
        transform="translate(10, 10)"
      >
        T
      </text>
    </g>
    <g
      transform="translate(40, 0)"
    >
      <rect
        fill="MediumPurple"
        height={21}
        width={21}
        x={0}
        y={0}
      />
      <text
        dy=".25em"
        fill="DarkMagenta"
        textAnchor="middle"
        transform="translate(10, 10)"
      >
        C
      </text>
    </g>
    <g
      transform="translate(60, 0)"
    >
      <rect
        fill="LemonChiffon"
        height={21}
        width={21}
        x={0}
        y={0}
      />
      <text
        dy=".25em"
        fill="GoldenRod"
        textAnchor="middle"
        transform="translate(10, 10)"
      >
        G
      </text>
    </g>
    <g
      transform="translate(80, 0)"
    >
      <rect
        fill="LightBlue"
        height={21}
        width={21}
        x={0}
        y={0}
      />
      <text
        dy=".25em"
        fill="Blue"
        textAnchor="middle"
        transform="translate(10, 10)"
      >
        T
      </text>
    </g>
    <g
      transform="translate(100, 0)"
    >
      <rect
        fill="LightPink"
        height={21}
        width={21}
        x={0}
        y={0}
      />
      <text
        dy=".25em"
        fill="Red"
        textAnchor="middle"
        transform="translate(10, 10)"
      >
        A
      </text>
    </g>
    <g
      transform="translate(120, 0)"
    >
      <rect
        fill="LightPink"
        height={21}
        width={21}
        x={0}
        y={0}
      />
      <text
        dy=".25em"
        fill="Red"
        textAnchor="middle"
        transform="translate(10, 10)"
      >
        A
      </text>
    </g>
    <g
      transform="translate(140, 0)"
    >
      <rect
        fill="LightBlue"
        height={21}
        width={21}
        x={0}
        y={0}
      />
      <text
        dy=".25em"
        fill="Blue"
        textAnchor="middle"
        transform="translate(10, 10)"
      >
        T
      </text>
    </g>
    <g
      transform="translate(160, 0)"
    >
      <rect
        fill="LightBlue"
        height={21}
        width={21}
        x={0}
        y={0}
      />
      <text
        dy=".25em"
        fill="Blue"
        textAnchor="middle"
        transform="translate(10, 10)"
      >
        T
      </text>
    </g>
    <g
      transform="translate(180, 0)"
    >
      <rect
        fill="LemonChiffon"
        height={21}
        width={21}
        x={0}
        y={0}
      />
      <text
        dy=".25em"
        fill="GoldenRod"
        textAnchor="middle"
        transform="translate(10, 10)"
      >
        G
      </text>
    </g>
    <g
      transform="translate(200, 0)"
    >
      <rect
        fill="MediumPurple"
        height={21}
        width={21}
        x={0}
        y={0}
      />
      <text
        dy=".25em"
        fill="DarkMagenta"
        textAnchor="middle"
        transform="translate(10, 10)"
      >
        C
      </text>
    </g>
    <g
      transform="translate(220, 0)"
    >
      <rect
        fill="LightPink"
        height={21}
        width={21}
        x={0}
        y={0}
      />
      <text
        dy=".25em"
        fill="Red"
        textAnchor="middle"
        transform="translate(10, 10)"
      >
        A
      </text>
    </g>
    <g
      transform="translate(0, 20)"
    >
      <rect
        fill="MediumPurple"
        height={21}
        width={21}
        x={0}
        y={0}
      />
      <text
        dy=".25em"
        fill="DarkMagenta"
        textAnchor="middle"
        transform="translate(10, 10)"
      >
        C
      </text>
    </g>
    <g
      transform="translate(20, 20)"
    >
      <rect
        fill="LightBlue"
        height={21}
        width={21}
        x={0}
        y={0}
      />
      <text
        dy=".25em"
        fill="Blue"
        textAnchor="middle"
        transform="translate(10, 10)"
      >
        T
      </text>
    </g>
    <g
      transform="translate(40, 20)"
    >
      <rect
        fill="MediumPurple"
        height={21}
        width={21}
        x={0}
        y={0}
      />
      <text
        dy=".25em"
        fill="DarkMagenta"
        textAnchor="middle"
        transform="translate(10, 10)"
      >
        C
      </text>
    </g>
    <g
      transform="translate(60, 20)"
    >
      <rect
        fill="LemonChiffon"
        height={21}
        width={21}
        x={0}
        y={0}
      />
      <text
        dy=".25em"
        fill="GoldenRod"
        textAnchor="middle"
        transform="translate(10, 10)"
      >
        G
      </text>
    </g>
    <g
      transform="translate(80, 20)"
    >
      <rect
        fill="LightBlue"
        height={21}
        width={21}
        x={0}
        y={0}
      />
      <text
        dy=".25em"
        fill="Blue"
        textAnchor="middle"
        transform="translate(10, 10)"
      >
        T
      </text>
    </g>
    <g
      transform="translate(100, 20)"
    >
      <rect
        fill="LightPink"
        height={21}
        width={21}
        x={0}
        y={0}
      />
      <text
        dy=".25em"
        fill="Red"
        textAnchor="middle"
        transform="translate(10, 10)"
      >
        A
      </text>
    </g>
    <g
      transform="translate(120, 20)"
    >
      <rect
        fill="LightPink"
        height={21}
        width={21}
        x={0}
        y={0}
      />
      <text
        dy=".25em"
        fill="Red"
        textAnchor="middle"
        transform="translate(10, 10)"
      >
        A
      </text>
    </g>
    <g
      transform="translate(140, 20)"
    >
      <rect
        fill="LightBlue"
        height={21}
        width={21}
        x={0}
        y={0}
      />
      <text
        dy=".25em"
        fill="Blue"
        textAnchor="middle"
        transform="translate(10, 10)"
      >
        T
      </text>
    </g>
    <g
      transform="translate(160, 20)"
    >
      <rect
        fill="LemonChiffon"
        height={21}
        width={21}
        x={0}
        y={0}
      />
      <text
        dy=".25em"
        fill="GoldenRod"
        textAnchor="middle"
        transform="translate(10, 10)"
      >
        G
      </text>
    </g>
    <g
      transform="translate(180, 20)"
    >
      <rect
        fill="LemonChiffon"
        height={21}
        width={21}
        x={0}
        y={0}
      />
      <text
        dy=".25em"
        fill="GoldenRod"
        textAnchor="middle"
        transform="translate(10, 10)"
      >
        G
      </text>
    </g>
    <g
      transform="translate(200, 20)"
    >
      <rect
        fill="MediumPurple"
        height={21}
        width={21}
        x={0}
        y={0}
      />
      <text
        dy=".25em"
        fill="DarkMagenta"
        textAnchor="middle"
        transform="translate(10, 10)"
      >
        C
      </text>
    </g>
    <g
      transform="translate(220, 20)"
    >
      <rect
        fill="MediumPurple"
        height={21}
        width={21}
        x={0}
        y={0}
      />
      <text
        dy=".25em"
        fill="DarkMagenta"
        textAnchor="middle"
        transform="translate(10, 10)"
      >
        C
      </text>
    </g>
    <g
      transform="translate(0, 40)"
    >
      <rect
        fill="LemonChiffon"
        height={21}
        width={21}
        x={0}
        y={0}
      />
      <text
        dy=".25em"
        fill="GoldenRod"
        textAnchor="middle"
        transform="translate(10, 10)"
      >
        G
      </text>
    </g>
    <g
      transform="translate(20, 40)"
    >
      <rect
        fill="LightBlue"
        height={21}
        width={21}
        x={0}
        y={0}
      />
      <text
        dy=".25em"
        fill="Blue"
        textAnchor="middle"
        transform="translate(10, 10)"
      >
        T
      </text>
    </g>
    <g
      transform="translate(40, 40)"
    >
      <rect
        fill="MediumPurple"
        height={21}
        width={21}
        x={0}
        y={0}
      />
      <text
        dy=".25em"
        fill="DarkMagenta"
        textAnchor="middle"
        transform="translate(10, 10)"
      >
        C
      </text>
    </g>
    <g
      transform="translate(60, 40)"
    >
      <rect
        fill="LemonChiffon"
        height={21}
        width={21}
        x={0}
        y={0}
      />
      <text
        dy=".25em"
        fill="GoldenRod"
        textAnchor="middle"
        transform="translate(10, 10)"
      >
        G
      </text>
    </g>
    <g
      transform="translate(80, 40)"
    >
      <rect
        fill="LightBlue"
        height={21}
        width={21}
        x={0}
        y={0}
      />
      <text
        dy=".25em"
        fill="Blue"
        textAnchor="middle"
        transform="translate(10, 10)"
      >
        T
      </text>
    </g>
    <g
      transform="translate(100, 40)"
    >
      <rect
        fill="MediumPurple"
        height={21}
        width={21}
        x={0}
        y={0}
      />
      <text
        dy=".25em"
        fill="DarkMagenta"
        textAnchor="middle"
        transform="translate(10, 10)"
      >
        C
      </text>
    </g>
    <g
      transform="translate(120, 40)"
    >
      <rect
        fill="LightPink"
        height={21}
        width={21}
        x={0}
        y={0}
      />
      <text
        dy=".25em"
        fill="Red"
        textAnchor="middle"
        transform="translate(10, 10)"
      >
        A
      </text>
    </g>
    <g
      transform="translate(140, 40)"
    >
      <rect
        fill="LightPink"
        height={21}
        width={21}
        x={0}
        y={0}
      />
      <text
        dy=".25em"
        fill="Red"
        textAnchor="middle"
        transform="translate(10, 10)"
      >
        A
      </text>
    </g>
    <g
      transform="translate(160, 40)"
    >
      <rect
        fill="LightBlue"
        height={21}
        width={21}
        x={0}
        y={0}
      />
      <text
        dy=".25em"
        fill="Blue"
        textAnchor="middle"
        transform="translate(10, 10)"
      >
        T
      </text>
    </g>
    <g
      transform="translate(180, 40)"
    >
      <rect
        fill="LemonChiffon"
        height={21}
        width={21}
        x={0}
        y={0}
      />
      <text
        dy=".25em"
        fill="GoldenRod"
        textAnchor="middle"
        transform="translate(10, 10)"
      >
        G
      </text>
    </g>
    <g
      transform="translate(200, 40)"
    >
      <rect
        fill="MediumPurple"
        height={21}
        width={21}
        x={0}
        y={0}
      />
      <text
        dy=".25em"
        fill="DarkMagenta"
        textAnchor="middle"
        transform="translate(10, 10)"
      >
        C
      </text>
    </g>
    <g
      transform="translate(220, 40)"
    >
      <rect
        fill="LightBlue"
        height={21}
        width={21}
        x={0}
        y={0}
      />
      <text
        dy=".25em"
        fill="Blue"
        textAnchor="middle"
        transform="translate(10, 10)"
      >
        T
      </text>
    </g>
  </g>
</svg>
`;
